src/snowflake/snowpark/_internal/temp_table_auto_cleaner.py

```diff
@@ -76,7 +79,8 @@ def drop_table(self, name: str) -> None:
             # TODO SNOW-1556553: Remove this workaround once multi-threading of Snowpark session is supported
             with self.session._conn._conn.cursor() as cursor:
                 cursor.execute(
-                    f"drop table if exists {name} /* internal query to drop unused temp table */"
+                    f"drop table if exists {name} /* internal query to drop unused temp table */",
+                    _statement_params={DROP_TABLE_STATEMENT_PARAM_NAME: True},
```

Comment:
Let's also log the table name here for easy lookup

Reply:
done

Reply:
since you removed the original true false, want to double confirm that we are basically rely on wether this statement parameter exists to check whether this table is auto dropped or not, right?

Reply:
yes